src/lib.rs

```diff
@@ -423,6 +423,56 @@ impl Build {
                 configure.arg("-D__STDC_NO_ATOMICS__");
             }
 
+            if target.contains("wasi") {
+                configure.args([
+                    // Termios isn't available whatsoever on WASM/WASI so we disable that
+                    // fatal error: 'termios.h' file not found
+                    "no-ui-console",
+                    // WASI doesn't support UNIX sockets so we preemptively disable it
+                    "no-sock",
+                    // WASI doesn't have a concept of syslog, so we disable it
+                    // fatal error: 'syslog.h' file not found
+                    "-DNO_SYSLOG",
+                    // wasm lacks signal support; to enable minimal signal emulation, compile with
+                    // -D_WASI_EMULATED_SIGNAL and link with -lwasi-emulated-signal
+                    "-D_WASI_EMULATED_SIGNAL",
+                    // WASI lacks process-associated clocks; to enable emulation of the `times` function using the wall
+                    // clock, which isn't sensitive to whether the program is running or suspended, compile with
+                    // -D_WASI_EMULATED_PROCESS_CLOCKS and link with -lwasi-emulated-process-clocks
+                    "-D_WASI_EMULATED_PROCESS_CLOCKS",
+                    // WASI lacks a true mmap; to enable minimal mmap emulation, compile
+                    // with -D_WASI_EMULATED_MMAN and link with -lwasi-emulated-mman
+                    "-D_WASI_EMULATED_MMAN",
+                    // WASI lacks process identifiers; to enable emulation of the `getpid` function using a
+                    // placeholder value, which doesn't reflect the host PID of the program, compile with
+                    // -D_WASI_EMULATED_GETPID and link with -lwasi-emulated-getpid
+                    "-D_WASI_EMULATED_GETPID",
+                    // ----------
+                    // The following arguments are taken from:
+                    // https://github.com/wapm-packages/OpenSSL/blob/master/build.sh#L21
+                    // Since this is a successful and reliable source for configuring WASM/I builds.
+                    // I will try to explain those to the best of my ability though.
+                    // ----------
+                    // WASI doesn't support (p)threads. Disabling preemptively.
+                    "no-threads",
+                    // WASI/WASM aren't really friends with ASM, so we disable it as well.
+                    "no-asm",
+                    // Disables the AFALG engine (AFALG-ENGine)
+                    // Since AFALG depends on `AF_ALG` support on the linux kernel side
+                    // it makes sense that we can't use it.
+                    "no-afalgeng",
+                    "-DOPENSSL_NO_AFALGENG=1",
+                    // ? Do we really want to always build statically for wasm? I guess yes?
```

Comment:
Yes, for wasm everything is always static right now. I don't know what effect this has on OpenSSL's build system though.

Reply:
Turns out providing `static` in this location as a cfg arg to OpenSSL prevents compilation as it would be provided *twice*. I removed it.